src/parallel/boundaries/BoundaryUtils.cpp

```diff
@@ -0,0 +1,304 @@
+                                      double nextStepVelAdjustment) {
+  const int ls1dim = convertDimensionToLS1Dims(dimension);
+  const int direction = findSign(dimension);
+  const double newPos =
+      molecule.r(ls1dim) +
+      (timestepLength * (molecule.v(ls1dim) + nextStepVelAdjustment));
+  if (newPos <= regionBegin[ls1dim] && direction < 0)
+    return true;
+  if (newPos >= regionEnd[ls1dim] && direction > 0)
+    return true;
+  return false;
+}
+
+std::tuple<std::array<double, 3>, std::array<double, 3>>
+BoundaryUtils::getOuterBuffer(const std::array<double, 3> givenRegionBegin,
+                              const std::array<double, 3> givenRegionEnd,
+                              DimensionType dimension, double *regionWidth) {
+  std::array<double, 3> returnRegionBegin, returnRegionEnd;
+  for (int i = 0; i < 3; i++) {
+    returnRegionBegin[i] = givenRegionBegin[i];
+    returnRegionEnd[i] = givenRegionEnd[i];
+  }
+
+  const int dimensionLS1 = convertDimensionToLS1Dims(dimension);
+
+  // find the two dimensions that are not being considered
+  const int extraDim1 = dimensionLS1 == 0 ? 1 : 0;
+  const int extraDim2 = dimensionLS1 == 2 ? 1 : 2;
+
+  // extend the extra dimensions to cover all ghost areas
+  returnRegionBegin[extraDim1] =
+      returnRegionBegin[extraDim1] - regionWidth[extraDim1];
+  returnRegionEnd[extraDim1] =
+      returnRegionEnd[extraDim1] + regionWidth[extraDim1];
+
+  returnRegionBegin[extraDim2] =
+      returnRegionBegin[extraDim2] - regionWidth[extraDim2];
+  returnRegionEnd[extraDim2] =
+      returnRegionEnd[extraDim2] + regionWidth[extraDim2];
+
+  switch (dimension) // can be done with findsign() too, but this is clearer
+  {
+  // in positive case, move the box begin to edge of domain, and box end to
+  // beyond
+  case DimensionType::POSX:
+  case DimensionType::POSY:
+  case DimensionType::POSZ:
+    returnRegionBegin[dimensionLS1] = returnRegionEnd[dimensionLS1];
+    returnRegionEnd[dimensionLS1] =
+        returnRegionBegin[dimensionLS1] + regionWidth[dimensionLS1];
+    break;
+
+  // in negative case, move the box end to edge of domain, and box begin to
+  // beyond
+  case DimensionType::NEGX:
+  case DimensionType::NEGY:
+  case DimensionType::NEGZ:
+    returnRegionEnd[dimensionLS1] = returnRegionBegin[dimensionLS1];
+    returnRegionBegin[dimensionLS1] =
+        returnRegionEnd[dimensionLS1] - regionWidth[dimensionLS1];
+    break;
+
+  default:
+    Log::global_log->error()
+        << "Invalid dimension passed for inner buffer calculation" << std::endl;
+    mardyn_exit(1);
+  }
+  return std::make_tuple(returnRegionBegin, returnRegionEnd);
+}
+
+bool BoundaryUtils::isNearRel(double a, double b,
+                              double maxRelativeDifference) {
+  const auto greaterNumber = std::max(std::abs(a), std::abs(b));
+  const auto absoluteDifference = maxRelativeDifference * greaterNumber;
+  const auto diff = std::abs(a - b);
+  return diff <= absoluteDifference;
+}
```

Comment:
this function has nothing to do with `BoundaryUtils` and should be moved to `Math.h`.